lib/stripe/converter.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Stripe.Converter do
+
+  @doc """
+  Takes the module (e.g. `Stripe.Card`) and the response from Stripe and
+  returns a struct (e.g. `%Stripe.Card{}`) containing the Stripe response.
+  """
+  @spec stripe_map_to_struct(module, %{String.t => any}) :: struct
+  def stripe_map_to_struct(module, response) do
+    [_|struct_keys] = Map.keys(module.__struct__)
+
+    processed_map =
+      Enum.reduce(struct_keys, %{}, fn key, acc ->
+        value =
+          fetch_value(response, key)
+          |> convert_value()
+
+        value = Map.get(module.relationships, key)
+        |> build_struct(value)
+
+        Map.put(acc, key, value)
+      end)
+
+    struct(module, processed_map)
+  end
+
+  defp build_struct(nil, value), do: value
+  defp build_struct(_module, nil), do: nil
+  defp build_struct(DateTime, value) do
+    {:ok, value} = DateTime.from_unix(value)
+    value
+  end
+  defp build_struct(module, value) when is_map(value) do
+    stripe_map_to_struct(module, value)
+  end
+
+  defp fetch_value(response, key) do
+    case Map.fetch(response, key) do
+      {:ok, value} -> value
+      :error -> Map.get(response, convert_key(key))
+    end
+  end
+
+  defp convert_key(key) when is_atom(key), do: to_string(key)
+  defp convert_key(key) when is_binary(key), do: String.to_atom(key)
+
+  defp convert_value(value) when is_map(value) do
+    Enum.reduce(value, %{}, fn({key, value}, acc) ->
+      Map.put(acc, String.to_atom(key), convert_value(value))
```

Comment:
Ah! I see what you mean.

Two ideas:

1. I'm wondering if you could break it out so that top-level definitions are `String.to_existing_atom/1` and lower levels are taken without restriction (I believe that's the same logic, but with added complexity in the conversion implementation).
2. Is there a way to invert the conversion such that the input atom is converted to a string instead of vice-versa? Then you can just use the original input atom for the `Map.put/3`.

In theory the algorithm you designed is restricted enough that a user would have to intentionally misuse it (at great length, no less) to cause an atom table overflow, but I still prefer to be safer with atom conversions if we can.